internal/internal_worker_test.go

```diff
@@ -1161,6 +1161,57 @@ func testReplayWorkflowCancelChildWorkflowUnusualOrdering(ctx Context) error {
 	return nil
 }
 
+func (s *internalWorkerTestSuite) TestReplayWorkflowHistory_ChildWorkflowCancellation_WhenWorkflowCanceled() {
```

Comment:
Can you replicate the failure in an integration test? Seems straightforward enough to wait for a child workflow to start and then cancel its parent and confirm the parent panicked/WFT-failed before this fix.

(sorry, I just trust integration tests much more than manual history creation)

Reply:
No fair point I'll make an integration test, I will say this manual history is basically what the customers history looked like